modules/ocf_stats/manifests/prometheus.pp

```diff
@@ -20,20 +33,14 @@
         'job_name'        => 'node',
         'scrape_interval' => '5s',
         'scrape_timeout'  => '5s',
-        'static_configs'  => [
+
+        'file_sd_configs' => [
           {
-            'targets' => [
-              # A list of all hosts to scrape.
-              # We should probably figure out a better way of enumerating all the
-              # hosts.
-              'leprosy:9100',
-            ],
+            'files'            => [ '/var/local/prometheus-nodes.json' ],
```

Comment:
So what happens if we puppet a new machine at 12:04 (assuming you keep the cron job to update the nodes at the 3rd minute of every hour)? The file `/var/local/prometheus-nodes.json` doesn't exist. How will prometheus handle this?

Reply:
Not sure what you mean? This file will always exist. If we puppet a machine at the 4 minute mark, we'll have to either wait an hour for `gen-prometheus-nodes` to run again, or just manually run it ourselves.

Reply:
This isn't true any more with the exec resource on bootstrapping, right?

Reply:
Yee, he fixed it in a later commit after we chatted on irc